### PR TITLE
Re-enable pua checks for emoji, require legacy pua.

### PR DESCRIPTION
**Pull request by dougfelt**
Re-enable pua checks for emoji, require legacy pua.

Github has a bug where it doesn't refresh the initial text for a pull request...

### CODE REVIEW

**Comment by dougfelt** [comment]
ping

File: nototools/noto_lint.py [hunk 1]
```diff
@@ -585,6 +586,10 @@ def _cjk_set():
         ranges = unicode_data._parse_code_ranges(noto_data.CJK_RANGES_TXT)
         return code_range_to_set(ranges) & unicode_data.defined_characters()
 
+    def _emoji_pua_set():
+        """Returns the legacy PUA characters required for Android emoji."""
+        return lint_config.parse_int_ranges('fe4e5-fe4ee fe82c fe82e-fe837')
```
Comment (roozbehp) [inline nototools/noto_lint.py]: For codepoints, always use uppercase hex digits.

File: nototools/noto_lint.py [hunk 2]
```diff
@@ -855,13 +860,10 @@ def check_name_table():
...
 
-    def _check_needed_chars(cmap, char_filter):
-        # TODO(roozbeh): check the glyph requirements for controls specified at
-        # https://www.microsoft.com/typography/otspec/recom.htm
-
+    def _get_script_required(cmap):
         needed_chars = set()
         if font_props.script == "Qaae":  # Emoji
-            needed_chars = set()  # TODO: Check emoji coverage
+            needed_chars = _emoji_pua_set()  # legacy PUA for android emoji
```
Comment (roozbehp) [inline nototools/noto_lint.py]: I would keep the existing TODO to actually check for the rest of emoji characters to be there too.

File: nototools/noto_lint.py [hunk 3]
```diff
@@ -988,13 +998,26 @@ def check_cmap_table():
             _check_needed_chars(cmap, tests.get_filter('cmap/script_required'))
 
         if tests.check('cmap/private_use'):
-            privates_in_cmap = {char for char in cmap
-                                if unicode_data.is_private_use(char)}
+            needed_chars = _get_script_required(cmap)
+            pua_filter = tests.get_filter('cmap/private_use')
+            if pua_filter:
+              pua_filter = pua_filter[1].accept
```
Comment (roozbehp) [inline nototools/noto_lint.py]: Wrong number of spaces for indent?

File: nototools/noto_lint.py [hunk 4]
```diff
@@ -988,13 +998,26 @@ def check_cmap_table():
...
+            def check_pua(char):
```
Comment (roozbehp) [inline nototools/noto_lint.py]: It's hard to understand what check_pua() is supposed to do based on its current name.

**Comment by dougfelt** [comment]
please take another look

**Comment by roozbehp** [comment]
LGTM

**Pull request merged**